src/redprl/categorical_judgment.sml

```diff
@@ -1,42 +1,70 @@
 structure RedPrlCategoricalJudgment :
 sig
   datatype ('sym, 'a) redprl_jdg =
-     EQ of ('a * 'a) * 'a
-   | TRUE of 'a
-   | EQ_TYPE of 'a * 'a
-   | SYNTH of 'a
+    (* `EQ_TYPE ((a, b), k)`:
+     *   `a` and `b` are equal types, taking into account the additional structures specified by `k`.
+     *   (That is, `EQ_TYPE ((a, b), KAN)` means they are equally Kan, not just having the same PER.)
+     *   The realizer is `TV` of sort `TRIV`.
+     *)
+     EQ_TYPE of ('a * 'a) * RedPrlKind.t
+   (* `EQ ((m, n), (a, k))`:
+    *   `EQ_TYPE ((a, a), k)` and `m` and `n` are related by the PER associated with `a`.
+     *   The realizer is `TV` of sort `TRIV`.
+    *)
+   | EQ of ('a * 'a) * ('a * RedPrlKind.t)
+    (* `TRUE (a, k)`:
+     *   `EQ_TYPE ((a, a), k)` and there exists a term `m` such that
+     *   `EQ ((m, m), (a, k))` is provable.
+     *   The realizer is such an `m` of sort `EXP`.
+     *)
+   | TRUE of ('a * RedPrlKind.t)
+    (* `SYNTH (m, k)`:
+     *   There exists a term `a` such that `EQ ((m, m), (a, k))` is provable.
+     *   Usually there is a canonical choice of `a`, but the canonicity is not required.
```

Comment:
Can we remove the sentence about canonical choices? I think that sentence makes sense, but it might be confusing for people looking at the documentation.